File: scratch/check_saphok_issues.js
const { createClient } = require('@supabase/supabase-js')
require('dotenv').config({ path: '.env.local' })

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL
const supabaseKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY
const supabase = createClient(supabaseUrl, supabaseKey)

async function test() {
  const date = '2026-05-21' // หรือวันที่ล่าสุดในระบบ
  
  // 1. ดู assignments สะโพกของวันนี้
  const { data: assignments } = await supabase
    .from('production_assignments')
    .select('*')
    .eq('production_date', date)
    .eq('table_name', 'สะโพก')
  
  console.log('--- ASSIGNMENTS IN DB FOR สะโพก ---')
  console.log(assignments ? assignments.map(a => `${a.sku} (${a.sku_name}) -> Qty: ${a.target_quantity}, Worker: ${a.worker_name}, Period: ${a.period}, Channel: ${a.channel}`).join('\n') : 'none')
  
  // 2. ดูยอดสั่งซื้อวันนี้ของ Lotus, Makro, WM สำหรับสะโพก
  // โดยหาจาก master_productivity ก่อนว่า SKU ไหนอยู่สะโพก
  const { data: prodMaster } = await supabase
    .from('master_logic_calculation')
    .select('row_data')
    .eq('calculation_type', 'Mas Productivity')
  
  const saphokSkus = new Set()
  const skuInfo = {}
  for (const r of prodMaster || []) {
    const station = String(r.row_data['จุดงาน'] || '').trim()
    const sku = String(r.row_data['SAP'] || '').trim().replace(/^0+/, '')
    const name = String(r.row_data['ชื่อสินค้า'] || '').trim()
    if (station.includes('สะโพก')) {
      saphokSkus.add(sku)
      skuInfo[sku] = { name, station }
    }
  }
  
  console.log('\n--- SAPHOK SKUS IN PRODUCTIVITY MASTER ---')
  console.log(Array.from(saphokSkus).map(sku => `${sku}: ${skuInfo[sku].name} (${skuInfo[sku].station})`).join('\n'))

  // ดึง order ของวันนี้
  const channels = ['wet_market_orders', 'lotus_orders', 'makro_orders']
  console.log('\n--- TODAY ORDERS FOR SAPHOK SKUS ---')
  for (const table of channels) {
    const { data: orders } = await supabase
      .from(table)
      .select('sku, sku_name, quantity, upload_round')
      .eq('delivery_date', date)
    
    const filtered = (orders || []).filter(o => saphokSkus.has(o.sku.replace(/^0+/, '')))
    console.log(`> ${table}:`)
    if (filtered.length) {
      console.log(filtered.map(o => `  ${o.sku} (${o.sku_name}) -> Qty: ${o.quantity}, Round: ${o.upload_round}`).join('\n'))
    } else {
      console.log('  none')
    }
  }
}

test()

File: scratch/check_saphok_issues.py
import os
import json
import urllib.request
import urllib.parse

# Load env variables from .env.local
supabase_url = None
supabase_key = None

if os.path.exists('.env.local'):
    with open('.env.local', 'r', encoding='utf-8') as f:
        for line in f:
            if line.strip().startswith('NEXT_PUBLIC_SUPABASE_URL='):
                supabase_url = line.strip().split('=', 1)[1]
            elif line.strip().startswith('NEXT_PUBLIC_SUPABASE_ANON_KEY='):
                supabase_key = line.strip().split('=', 1)[1]

if not supabase_url or not supabase_key:
    print("Failed to load Supabase env variables.")
    exit(1)

def http_get(url):
    req = urllib.request.Request(url)
    req.add_header('apikey', supabase_key)
    req.add_header('Authorization', f'Bearer {supabase_key}')
    req.add_header('Content-Type', 'application/json')
    try:
        with urllib.request.urlopen(req) as response:
            return json.loads(response.read().decode('utf-8'))
    except Exception as e:
        print(f"Error requesting {url}: {e}")
        return []

date = '2026-05-21'

# 1. Query assignments for sa-phok
url_assign = f"{supabase_url}/rest/v1/production_assignments?production_date=eq.{date}&table_name=eq.สะโพก"
assignments = http_get(url_assign)

print("--- ASSIGNMENTS IN DB FOR สะโพก ---")
if assignments:
    for a in assignments:
        print(f"{a.get('sku')} ({a.get('sku_name')}) -> Qty: {a.get('target_quantity')}, Worker: {a.get('worker_name')}, Period: {a.get('period')}, Channel: {a.get('channel')}")
else:
    print("none")

# 2. Query master productivity to find sa-phok SKUs
url_master = f"{supabase_url}/rest/v1/master_logic_calculation?calculation_type=eq.Mas Productivity&order=uploaded_at.desc&limit=500"
master_rows = http_get(url_master)

saphok_skus = set()
sku_info = {}
for row in master_rows:
    row_data = row.get('row_data', {})
    station = str(row_data.get('จุดงาน', '')).strip()
    sku = str(row_data.get('SAP', '')).strip().lstrip('0')
    name = str(row_data.get('ชื่อสินค้า', '')).strip()
    if 'สะโพก' in station:
        saphok_skus.add(sku)
        sku_info[sku] = {'name': name, 'station': station}

print("\n--- SAPHOK SKUS IN PRODUCTIVITY MASTER ---")
for sku in sorted(list(saphok_skus)):
    print(f"{sku}: {sku_info[sku]['name']} ({sku_info[sku]['station']})")

# 3. Query today's orders
channels = ['wet_market_orders', 'lotus_orders', 'makro_orders']
print("\n--- TODAY ORDERS FOR SAPHOK SKUS ---")
for table in channels:
    url_orders = f"{supabase_url}/rest/v1/{table}?delivery_date=eq.{date}"
    orders = http_get(url_orders)
    
    filtered = [o for o in orders if str(o.get('sku', '')).strip().lstrip('0') in saphok_skus]
    print(f"> {table}:")
    if filtered:
        for o in filtered:
            print(f"  {o.get('sku')} ({o.get('sku_name')}) -> Qty: {o.get('quantity')}, Round: {o.get('upload_round')}")
    else:
        print("  none")
